pkg/controller/hibernation/hibernation_controller.go

```diff
@@ -178,7 +178,11 @@ func (r *hibernationReconciler) Reconcile(ctx context.Context, request reconcile
 
 	// If cluster is already deleted, skip any processing
 	if !cd.DeletionTimestamp.IsZero() {
-		return reconcile.Result{}, nil
+		cd.Status.PowerState = hivev1.ClusterPowerStateUnknown
+		msg := "ClusterDeployment has been marked for deletion"
+		r.setCDCondition(cd, hivev1.ClusterHibernatingCondition, hivev1.HibernatingReasonClusterDeploymentDeleted, msg, corev1.ConditionUnknown, cdLog)
+		r.setCDCondition(cd, hivev1.ClusterReadyCondition, hivev1.ReadyReasonClusterDeploymentDeleted, msg, corev1.ConditionUnknown, cdLog)
+		return reconcile.Result{}, r.updateClusterDeploymentStatus(cd, cdLog)
```

Comment:
You should add a check here to see if an update is needed.
If a cluster is already marked for deletion and we have updated the status on CD in a previous reconcile, we do not need to call an `Update` again.